### ISSUE TITLE
Not understanding why custom validator shows error at odd time (Rails5 / Devise)

### ISSUE THREAD

**Issue by isuPatches** [DavyJonesLocker/client_side_validations#674]
For some reason, I'm seeing the custom validator take affect, but not until I have posted the form which seems really odd to me.  Am I missing something obvious?

Version of ClientSideValidations:

```
GIT
  remote: https://github.com/DavyJonesLocker/client_side_validations.git
  revision: 01a0b88d058080d60d6245711bbd6ff964663efe
  branch: rails5
  specs:
    client_side_validations (5.0.0)
      jquery-rails (~> 4.0, >= 4.0.1)
      js_regex (~> 1.0, >= 1.0.9)
      rails (>= 5.0.0.rc2, < 5.1.0)
```

Version of Rails:

` rails (5.0.0.1)`

Code snippet from your model of the validations:

```
class AdminUser < ActiveRecord::Base
  
  validates_password_complexity :password

  devise :confirmable, :database_authenticatable, :recoverable, :registerable,
         :rememberable, :trackable, :timeoutable, :lockable, :timeoutable, :validatable

end
```

The form code from your template:

```
<h2>Sign up</h2>

<%= form_for(resource, validate: true, as: resource_name, url: registration_path(resource_name)) do |f| %>
  <%= devise_error_messages! %>

  <div class="field">
    <%= f.label :email %><br />
    <%= f.email_field :email, autofocus: true %>
  </div>

  <div class="field">
    <%= f.label :password %>
    <% if @minimum_password_length %>
    <em>(<%= @minimum_password_length %> characters minimum)</em>
    <% end %><br />
    <%= f.password_field :password, autocomplete: "off" %>
  </div>

  <div class="field">
    <%= f.label :password_confirmation %><br />
    <%= f.password_field :password_confirmation, autocomplete: "off" %>
  </div>

  <div class="actions">
    <%= f.submit "Sign up" %>
  </div>
<% end %>

<%= render "admin_users/shared/links" %>
```
The resulting HTML along with the script tag:

Before I hit the sign-up button:

```
<!DOCTYPE html>
<html>
  <head>
    <title>RockThePatchv4</title>
    <meta name="csrf-param" content="authenticity_token" />
<meta name="csrf-token" content="Y8rZi8AJLDNWLVqsNZCaslboNgJLQmIsYFW2a6dPzLwP39pdtEfwqa5xeqiBuazgshjr+fQ3gRWDgfKF8HiLVQ==" />

    <link rel="stylesheet" media="all" href="/assets/base/_colors.self-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/base/_fonts.self-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/base/_globals.self-88463f8a2cfcccbe6eaf06c0c01de27ba7532b188301ad3451c98bb5d9d2ff47.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/base/_mixins.self-e98949a9e163a2985d229801711b65bd78dae878894386ddd23c6023a4d677c8.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/base/_variables.self-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/bootstrap/_overrides.self-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/forms/base_form.self-be812009a8e681a653f9697fad8c6c6acc07d13db44eb03c6ec617820256da45.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/forms/forgot-password-form.self-ebf93e496b7b614a16d6c8a6cf4a72675dd06838514d5e6584060adfff773cef.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/forms/forgot-username-form.self-49ded2553544e535083b910890ab3d9a5ff1d179e3184028df5e1883fbd09c68.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/forms/login_form.self-1c96046a0a18604e8a6de66f7427741dfc96ae9283d04149a1b553a189122f0f.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/forms/password-reset-form.self-946eaad4444cad84fe50883d3fe936f813b3cc35402fe5c442db840df71a9ed2.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/forms/registration_form.self-90adc4d78623a7c3a123a804a0b10d2417ecd230b3b4243a60ca12ce134fc7a6.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/forms/resend-confirmation-form.self-ffa48f8472aea7d24ebbfd5de0905a316cfa816468c5c40b4b67d659b54def77.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/layouts/_container.self-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/layouts/_footer.self-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/layouts/_header.self-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/modules/flash.self-3613e68ebd86c235b5246a09e5829be3fe4f5483f78a0ee0dd772619e382aba7.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/application.self-5ba6c6379b450e21bcb99669ac7e99dd4ff1d2e813d05d3e0c08d971615b86b7.css?body=1" data-turbolinks-track="reload" />
    <script src="/assets/jquery.self-bd7ddd393353a8d2480a622e80342adf488fb6006d667e8b42e4c0073393abee.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/jquery_ujs.self-784a997f6726036b1993eb2217c9cb558e1cbb801c6da88105588c56f13b466a.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/rails.validations.self-98c2768ab2a3093d0e5b8cb558b96a31dca03f2bbc82b6f8682d3830bebb720a.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/rails.validations.password_complexity.self-fe342f3d1e341c028bbeb1dd06173371eae2d8e0c93688c3b5020b02dae1c1a0.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/rails.validations.simple_form.self-e781a2ebba41b673c8d3f2e4b5c38504fca848d01bcf75fd6ec09fad2bb0dbee.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/bootstrap/transition.self-6ad2488465135ab731a045a8ebbe3ea2fc501aed286042496eda1664fdd07ba9.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/bootstrap/alert.self-742145c5bb847aafdadc6e339be795628f8bc25f177e851f03a8c42278eb0312.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/bootstrap/button.self-126ac9bf0e7f2d8568f8da3a00fd5f0fac6eae0946331003370161fbf8d7975e.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/bootstrap/carousel.self-e47323f363ceb3dc0bdbce05e36e709ed428e339833a41140a85cb0af24b8127.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/bootstrap/collapse.self-2eb697f62b587bb786ff940d82dd4be88cdeeaf13ca128e3da3850c5fcaec301.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/bootstrap/dropdown.self-561cca1cbaf67474e01e9536f106bad541594860a6df997004591c1c1957a147.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/bootstrap/modal.self-3e78617ade5663314b7ee0ea10375a5b34d59ffbade44939e3f2a4e4ef2019b3.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/bootstrap/tab.self-5bf7078b682f8b131332eefa46b45fa5eff2eca745fc0d03e2991450888f7c28.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/bootstrap/affix.self-6d6f1a7fc5c8aabf3547fa1b794fab6268f54bc55ad815e55873c71f52513517.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/bootstrap/scrollspy.self-969f3c5f48cdf1e439c7fa1154c13b948715f5c689f87837c0b64521d3b46ef6.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/bootstrap/tooltip.self-05afb177e08f98997ccfc84fa08a215e4b27d48d5fe4d049080675e9dffd8199.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/bootstrap/popover.self-0aa93860b59fe7393f1dd490f54b3cb994f9d6155adffce034d4e14ae361b041.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/bootstrap-sprockets.self-fbfa5ad7d9aa0afe439ec4ff3883acc4cb92b62cb67c40d674320c9aa1d4642d.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/application.self-4d37d7bcf4b0b32507438063ae47ba88dd0178863f7c7a2c7125b221eefea772.js?body=1" data-turbolinks-track="reload"></script>
  </head>

  <body>
    <div class="center-text">
</div>

    <h2>Sign up</h2>

<form data-validate="true" novalidate="novalidate" class="new_admin_user" id="new_admin_user" action="/authorized" accept-charset="UTF-8" method="post"><input name="utf8" type="hidden" value="&#x2713;" /><input type="hidden" name="authenticity_token" value="7C5KhsWC6UozPVfXsAnC84tleQ0VGK+8iwTdUqUo1zafSdSWTjcPdceOTzIVQpJj8AM+X2novGDR0X//E7+J4g==" />
  

  <div class="field">
    <label for="admin_user_email">Email</label><br />
    <input autofocus="autofocus" type="email" value="" name="admin_user[email]" id="admin_user_email" />
  </div>

  <div class="field">
    <label for="admin_user_password">Password</label>
    <em>(8 characters minimum)</em>
    <br />
    <input autocomplete="off" type="password" name="admin_user[password]" id="admin_user_password" />
  </div>

  <div class="field">
    <label for="admin_user_password_confirmation">Password confirmation</label><br />
    <input autocomplete="off" type="password" name="admin_user[password_confirmation]" id="admin_user_password_confirmation" />
  </div>

  <div class="actions">
    <input type="submit" name="commit" value="Sign up" data-disable-with="Sign up" />
  </div>
</form><script>
//<![CDATA[
if(window.ClientSideValidations===undefined)window.ClientSideValidations={};window.ClientSideValidations.disabled_validators=["uniqueness"];window.ClientSideValidations.number_format={"separator":".","delimiter":","};if(window.ClientSideValidations.patterns===undefined)window.ClientSideValidations.patterns = {};window.ClientSideValidations.patterns.numericality=/^(-|\+)?(?:\d+|\d{1,3}(?:\,\d{3})+)(?:\.\d*)?$/;if(window.ClientSideValidations.forms===undefined)window.ClientSideValidations.forms={};window.ClientSideValidations.forms['new_admin_user'] = {"type":"ActionView::Helpers::FormBuilder","input_tag":"\u003cdiv class=\"form-error\"\u003e\u003cspan id=\"input_tag\" /\u003e\u003clabel for=\"\" class=\"message\"\u003e\u003c/label\u003e\u003c/div\u003e","label_tag":"\u003cdiv class=\"form-error\"\u003e\u003clabel id=\"label_tag\" /\u003e\u003c/div\u003e","validators":{}};
//]]>
</script>
  <a href="/authorized/login">Log in</a><br />



  <a href="/authorized/confirmation/new">Didn&#39;t receive confirmation instructions?</a><br />

  <a href="/authorized/unlock/new">Didn&#39;t receive unlock instructions?</a><br />



  </body>
</html>

```
<img width="286" alt="screen shot 2016-09-19 at 3 32 04 pm" src="https://cloud.githubusercontent.com/assets/4986380/18648077/9e4c6e08-7e7e-11e6-8400-1285af15e6d7.png">

After I hit the sign-up button:

```
<!DOCTYPE html>
<html>
  <head>
    <title>RockThePatchv4</title>
    <meta name="csrf-param" content="authenticity_token" />
<meta name="csrf-token" content="ImCA6xG9mBbyD39O+GLbEkXw0GS51iowjZf4pejP7lBOdYM9ZfNEjApTX0pMS+1AoQANnwajyQluQ7xLv/ipuQ==" />

    <link rel="stylesheet" media="all" href="/assets/base/_colors.self-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/base/_fonts.self-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/base/_globals.self-88463f8a2cfcccbe6eaf06c0c01de27ba7532b188301ad3451c98bb5d9d2ff47.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/base/_mixins.self-e98949a9e163a2985d229801711b65bd78dae878894386ddd23c6023a4d677c8.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/base/_variables.self-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/bootstrap/_overrides.self-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/forms/base_form.self-be812009a8e681a653f9697fad8c6c6acc07d13db44eb03c6ec617820256da45.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/forms/forgot-password-form.self-ebf93e496b7b614a16d6c8a6cf4a72675dd06838514d5e6584060adfff773cef.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/forms/forgot-username-form.self-49ded2553544e535083b910890ab3d9a5ff1d179e3184028df5e1883fbd09c68.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/forms/login_form.self-1c96046a0a18604e8a6de66f7427741dfc96ae9283d04149a1b553a189122f0f.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/forms/password-reset-form.self-946eaad4444cad84fe50883d3fe936f813b3cc35402fe5c442db840df71a9ed2.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/forms/registration_form.self-90adc4d78623a7c3a123a804a0b10d2417ecd230b3b4243a60ca12ce134fc7a6.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/forms/resend-confirmation-form.self-ffa48f8472aea7d24ebbfd5de0905a316cfa816468c5c40b4b67d659b54def77.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/layouts/_container.self-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/layouts/_footer.self-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/layouts/_header.self-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/modules/flash.self-3613e68ebd86c235b5246a09e5829be3fe4f5483f78a0ee0dd772619e382aba7.css?body=1" data-turbolinks-track="reload" />
<link rel="stylesheet" media="all" href="/assets/application.self-5ba6c6379b450e21bcb99669ac7e99dd4ff1d2e813d05d3e0c08d971615b86b7.css?body=1" data-turbolinks-track="reload" />
    <script src="/assets/jquery.self-bd7ddd393353a8d2480a622e80342adf488fb6006d667e8b42e4c0073393abee.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/jquery_ujs.self-784a997f6726036b1993eb2217c9cb558e1cbb801c6da88105588c56f13b466a.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/rails.validations.self-98c2768ab2a3093d0e5b8cb558b96a31dca03f2bbc82b6f8682d3830bebb720a.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/rails.validations.password_complexity.self-fe342f3d1e341c028bbeb1dd06173371eae2d8e0c93688c3b5020b02dae1c1a0.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/rails.validations.simple_form.self-e781a2ebba41b673c8d3f2e4b5c38504fca848d01bcf75fd6ec09fad2bb0dbee.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/bootstrap/transition.self-6ad2488465135ab731a045a8ebbe3ea2fc501aed286042496eda1664fdd07ba9.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/bootstrap/alert.self-742145c5bb847aafdadc6e339be795628f8bc25f177e851f03a8c42278eb0312.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/bootstrap/button.self-126ac9bf0e7f2d8568f8da3a00fd5f0fac6eae0946331003370161fbf8d7975e.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/bootstrap/carousel.self-e47323f363ceb3dc0bdbce05e36e709ed428e339833a41140a85cb0af24b8127.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/bootstrap/collapse.self-2eb697f62b587bb786ff940d82dd4be88cdeeaf13ca128e3da3850c5fcaec301.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/bootstrap/dropdown.self-561cca1cbaf67474e01e9536f106bad541594860a6df997004591c1c1957a147.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/bootstrap/modal.self-3e78617ade5663314b7ee0ea10375a5b34d59ffbade44939e3f2a4e4ef2019b3.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/bootstrap/tab.self-5bf7078b682f8b131332eefa46b45fa5eff2eca745fc0d03e2991450888f7c28.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/bootstrap/affix.self-6d6f1a7fc5c8aabf3547fa1b794fab6268f54bc55ad815e55873c71f52513517.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/bootstrap/scrollspy.self-969f3c5f48cdf1e439c7fa1154c13b948715f5c689f87837c0b64521d3b46ef6.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/bootstrap/tooltip.self-05afb177e08f98997ccfc84fa08a215e4b27d48d5fe4d049080675e9dffd8199.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/bootstrap/popover.self-0aa93860b59fe7393f1dd490f54b3cb994f9d6155adffce034d4e14ae361b041.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/bootstrap-sprockets.self-fbfa5ad7d9aa0afe439ec4ff3883acc4cb92b62cb67c40d674320c9aa1d4642d.js?body=1" data-turbolinks-track="reload"></script>
<script src="/assets/application.self-4d37d7bcf4b0b32507438063ae47ba88dd0178863f7c7a2c7125b221eefea772.js?body=1" data-turbolinks-track="reload"></script>
  </head>

  <body>
    <div class="center-text">
</div>

    <h2>Sign up</h2>

<form data-validate="true" novalidate="novalidate" class="new_admin_user" id="new_admin_user" action="/authorized" accept-charset="UTF-8" method="post"><input name="utf8" type="hidden" value="&#x2713;" /><input type="hidden" name="authenticity_token" value="eVgbm75d4BBHawjgnqlsis6ELwsTj5jHIqhNGAN/JzIKP4WLNegGL7PYEAU74jwateJoWW9/ixt4fe+1teh55g==" />
      <div id="error_explanation">
      <h2>3 errors prohibited this admin user from being saved:</h2>
      <ul><li>Password must include at least one uppercase letter</li><li>Password must include at least one number</li><li>Email can&#39;t be blank</li></ul>
    </div>


  <div class="field">
    <div class="form-error"><label for="admin_user_email">Email</label></div><br />
    <div class="form-error"><input autofocus="autofocus" type="email" value="" name="admin_user[email]" id="admin_user_email" /><label for="admin_user_email" class="message">can't be blank</label></div>
  </div>

  <div class="field">
    <div class="form-error"><label for="admin_user_password">Password</label></div>
    <em>(8 characters minimum)</em>
    <br />
    <div class="form-error"><input autocomplete="off" type="password" name="admin_user[password]" id="admin_user_password" /><label for="admin_user_password" class="message">must include at least one uppercase letter</label></div>
  </div>

  <div class="field">
    <label for="admin_user_password_confirmation">Password confirmation</label><br />
    <input autocomplete="off" type="password" name="admin_user[password_confirmation]" id="admin_user_password_confirmation" />
  </div>

  <div class="actions">
    <input type="submit" name="commit" value="Sign up" data-disable-with="Sign up" />
  </div>
</form><script>
//<![CDATA[
if(window.ClientSideValidations===undefined)window.ClientSideValidations={};window.ClientSideValidations.disabled_validators=["uniqueness"];window.ClientSideValidations.number_format={"separator":".","delimiter":","};if(window.ClientSideValidations.patterns===undefined)window.ClientSideValidations.patterns = {};window.ClientSideValidations.patterns.numericality=/^(-|\+)?(?:\d+|\d{1,3}(?:\,\d{3})+)(?:\.\d*)?$/;if(window.ClientSideValidations.forms===undefined)window.ClientSideValidations.forms={};window.ClientSideValidations.forms['new_admin_user'] = {"type":"ActionView::Helpers::FormBuilder","input_tag":"\u003cdiv class=\"form-error\"\u003e\u003cspan id=\"input_tag\" /\u003e\u003clabel for=\"\" class=\"message\"\u003e\u003c/label\u003e\u003c/div\u003e","label_tag":"\u003cdiv class=\"form-error\"\u003e\u003clabel id=\"label_tag\" /\u003e\u003c/div\u003e","validators":{}};
//]]>
</script>
  <a href="/authorized/login">Log in</a><br />



  <a href="/authorized/confirmation/new">Didn&#39;t receive confirmation instructions?</a><br />

  <a href="/authorized/unlock/new">Didn&#39;t receive unlock instructions?</a><br />



  </body>
</html>
```

<img width="740" alt="screen shot 2016-09-19 at 3 33 53 pm" src="https://cloud.githubusercontent.com/assets/4986380/18648068/96b9b498-7e7e-11e6-9a38-c92616691999.png">

Initializer:

```
ActionView::Base.field_error_proc = Proc.new do |html_tag, instance|
  unless html_tag =~ /^<label/
    %{<div class="form-error">#{html_tag}<label for="#{instance.send(:tag_id)}" class="message">#{instance.error_message.first}</label></div>}.html_safe
  else
    %{<div class="form-error">#{html_tag}</div>}.html_safe
  end
end
```

The custom validator in initializers/validators:

```
class PasswordComplexityValidator < ActiveModel::Validator
  def validate(record)
    Rails.logger.debug record.password
    unless record.password.match(/[a-z]/)
      Rails.logger.debug 'Password must include at least one lowercase letter'
      record.errors[:password] << I18n.t('errors.must_include_at_least_one_lowercase_letter')
    end
    unless record.password.match(/[A-Z]/)
      Rails.logger.debug 'Password must include at least one uppercase letter'
      record.errors[:password] << I18n.t('errors.must_include_at_least_one_uppercase_letter')
    end
    unless record.password.match(/[0-9]/)
      Rails.logger.debug 'Password must include at least one number'
      record.errors[:password] << I18n.t('errors.must_include_at_least_one_number')
    end
  end
end

module ActiveModel::Validations::HelperMethods
  def validates_password_complexity(*attr_names)
    validates_with PasswordComplexityValidator, _merge_attributes(attr_names)
  end
end
```

**Comment by tagliala** [comment]
It looks like that CSV is not working at all, because it submitted the whole form before displaying errors. There is missing the javascript implementation of your custom validator (aka `rails.validations.password_complexity.js`).

Could you also take a look at the development console and check for errors?

**Comment by isuPatches** [comment]
I have not seen anything in my console.  I did upload:

https://github.com/isuPatches/ClientSideValidationSample which is a very basic app (sans secrets and database credits) with simple_form/devise/and client side validations that shows the JS not loading on http://localhost:3000/users/register.

I may be missing something really simple...thank you for helping!

**Comment by tagliala** [comment]
Apparently, you should implement your validator as an `ActiveModel::EachValidator` as described in [README](https://github.com/DavyJonesLocker/client_side_validations#local-validators)

```rb
class PasswordComplexityValidator < ActiveModel::EachValidator
  def validate_each(record, attr_name, value)
    Rails.logger.debug record.password
    unless value.match(/[a-z]/)
      Rails.logger.debug 'Password must include at least one lowercase letter'
      record.errors.add(attr_name, I18n.t('errors.must_include_at_least_one_lowercase_letter'))
    end
    unless value.match(/[A-Z]/)
      Rails.logger.debug 'Password must include at least one uppercase letter'
      record.errors.add(attr_name, I18n.t('errors.must_include_at_least_one_uppercase_letter'))
    end
    unless value.match(/[0-9]/)
      Rails.logger.debug 'Password must include at least one number'
      record.errors.add(attr_name, I18n.t('errors.must_include_at_least_one_number'))
    end
  end
end
```

Also, there seems to be no simple way of passing three different messages for the same validator to the client side validator, so you need to pass a hash of messages and then fix the server side validator and the javascript version accordingly

```yml
en:
  hello: "Hello world"
  errors:
    messages:
      password_complexity:
        must_include_at_least_one_lowercase_letter: Must include at least one lowercase letter
        must_include_at_least_one_uppercase_letter: Must include at least one uppercase letter
        must_include_at_least_one_number: Must include at least one number
```

```js
// The validator variable is a JSON Object
// The selector variable is a jQuery Object
window.ClientSideValidations.validators.local['password_complexity'] = function(element, options) {
    console.log(element.val());
    if (/[a-z]/.test(element.val())) {
        return options.message.must_include_at_least_one_lowercase_letter;
    } else if (/[A-Z]/.test(element.val())) {
        return options.message.must_include_at_least_one_uppercase_letter;
    } else if (/[0-9]/.test(element.val())) {
        return options.message.must_include_at_least_one_number;
    }
};
```

Also, custom validators are apparently run after the base validators, so you will get a surprise after confirming your password and this is really bad (but it is another issue, #655)

Hope it helps

Closing here but feel free to comment

**Comment by isuPatches** [comment]
Thank you thank you!  You've given me enough to get functional!
